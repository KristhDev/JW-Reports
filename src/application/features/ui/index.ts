export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';