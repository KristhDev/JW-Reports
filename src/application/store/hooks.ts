import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();