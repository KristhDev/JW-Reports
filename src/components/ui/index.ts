export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';