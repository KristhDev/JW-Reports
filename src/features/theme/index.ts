export { default as themeReducer, setTheme } from './slice';