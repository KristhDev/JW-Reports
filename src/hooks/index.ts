export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';