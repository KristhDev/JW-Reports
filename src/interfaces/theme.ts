export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}