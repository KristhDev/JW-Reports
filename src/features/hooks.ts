import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;