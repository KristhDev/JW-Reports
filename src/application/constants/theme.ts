import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];