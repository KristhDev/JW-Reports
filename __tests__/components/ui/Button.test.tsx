import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import { Button } from '../../../src/components/ui';

import { useTheme } from '../../../src/hooks';

import { darkColors } from '../../../src/theme';

const btnText = 'Test text';
const onPressMock = jest.fn();

jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <Button /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    beforeEach(() => {
        render(
            <Button
                icon={
                    <Icon
                        name="flask-outline"
                        size={ 20 }
                        testID="button-icon"
                    />
                }
                onPress={ onPressMock }
                text={ btnText }
            />
        );
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const text = screen.getByTestId('button-text');
        const icon = screen.getByTestId('button-icon');

        expect(text).toBeTruthy();
        expect(text.props.children).toBe(btnText);
        expect(icon).toBeTruthy();
    });

    it('should call onPress then press button', () => {
        const touchable = screen.getByTestId('button-touchable');
        fireEvent.press(touchable);

        expect(onPressMock).toHaveBeenCalledTimes(1);
    });
});