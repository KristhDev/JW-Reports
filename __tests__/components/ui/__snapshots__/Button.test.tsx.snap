// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Button /> component should to match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      false,
      Array [
        Object {
          "backgroundColor": "#C0A7E1",
          "borderRadius": 10,
        },
        undefined,
      ],
    ]
  }
  testID="button-touchable"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "#121212",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 2,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
      testID="button-text"
    >
      Test text
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="button-icon"
    >
      ï‹§
    </Text>
  </View>
</View>
`;
